contracts/RecoveryManager.sol

```diff
@@ -0,0 +1,88 @@
+pragma solidity 0.4.8;
+
+import "./IdentityManager.sol";
+import "./Proxy.sol";
+
+contract RecoveryManager {
+    IdentityManager identityManager;
+    
+    struct Delegate {
+        bool exists;
+        address proposedUserKey;
+    }
+    
+    struct RecoveryIdentity {
+        bool exists;
+        uint neededSignatures;
+        mapping (address => Delegate)  delegates;
+        address[] delegateAddresses; // needed for iteration of mapping
+    }
+    
+    //Create RecoveryManager w/ partner _identityManager
+    function RecoveryManager (address _identityManager) {
+        identityManager = IdentityManager(_identityManager);
+    }
+    
+    //Proxy address => RecoveryIdentity
+    mapping (address => RecoveryIdentity) recoveryIdentities;
+    
+    
+    modifier onlyDelegate(address identity) {
+        if (recoveryIdentities[identity].delegates[msg.sender].exists) _;
+        else throw;
+    }
+    
+    function createRecovery(address identity, address[] _delegates) {
+        RecoveryIdentity recovIden = recoveryIdentities[identity];
+        if (recovIden.exists) throw; //overwrite protection
+        
+        recovIden.exists = true;
+        recovIden.neededSignatures = _delegates.length/2 + 1;
+        
+        for (uint i = 0; i < _delegates.length; i++) {
```

Comment:
I think it would be relatively straightforward to have a similar `updateRecoveryDelegates` function with essentially the same signature.

It would just have to delete the delegates and set exists=false on all the delegate mappings.